tests/testthat/test-shinytest2-tm_g_bivariate.R

```diff
@@ -0,0 +1,146 @@
+          selected = "conc",
+          fixed = FALSE
+        )
+      ),
+      y = teal.transform::data_extract_spec(
+        dataname = "CO2",
+        select = teal.transform::select_spec(
+          label = "Select variable:",
+          choices = teal.transform::variable_choices(data[["CO2"]]),
+          selected = "uptake",
+          multiple = FALSE,
+          fixed = FALSE
+        )
+      ),
+      row_facet = teal.transform::data_extract_spec(
+        dataname = "CO2",
+        select = teal.transform::select_spec(
+          label = "Select variable:",
+          choices = teal.transform::variable_choices(data[["CO2"]]),
+          selected = "Type",
+          fixed = FALSE
+        )
+      ),
+      col_facet = teal.transform::data_extract_spec(
+        dataname = "CO2",
+        select = teal.transform::select_spec(
+          label = "Select variable:",
+          choices = teal.transform::variable_choices(data[["CO2"]]),
+          selected = "Treatment",
+          fixed = FALSE
+        )
+      ),
+      ggplot2_args = teal.widgets::ggplot2_args(
+        labs = list(subtitle = "Plot generated by Bivariate Module")
+      )
+    )
+  )
+}
+
+testthat::test_that("e2e - tm_g_bivariate: module is initialised with the specified defaults", {
+  skip_if_too_deep(5)
+
+  app <- app_driver_tm_g_bivariate()
+
+  app$expect_no_shiny_error()
+
+  testthat::expect_equal(app$get_active_module_input("x-dataset_CO2_singleextract-select"), "conc")
+  testthat::expect_equal(app$get_active_module_input("y-dataset_CO2_singleextract-select"), "uptake")
+  testthat::expect_true(app$get_active_module_input("facetting"))
+  testthat::expect_equal(app$get_active_module_input("row_facet-dataset_CO2_singleextract-select"), "Type")
+  testthat::expect_equal(app$get_active_module_input("col_facet-dataset_CO2_singleextract-select"), "Treatment")
+  testthat::expect_false(app$get_active_module_input("free_x_scales"))
+  testthat::expect_false(app$get_active_module_input("free_y_scales"))
+  testthat::expect_false(app$get_active_module_input("rotate_xaxis_labels"))
+  testthat::expect_false(app$get_active_module_input("swap_axes"))
+  testthat::expect_equal(app$get_active_module_input("ggtheme"), "gray")
+  testthat::expect_equal(app$get_active_module_input("alpha"), 0.5)
+  testthat::expect_equal(app$get_active_module_input("fixed_size"), 2)
+  testthat::expect_false(app$get_active_module_input("add_lines"))
+
+  app$stop()
+})
+
+testthat::test_that("e2e - tm_g_bivariate: facetting options are hidden when facet is toggled off", {
+  skip_if_too_deep(5)
+
+  app <- app_driver_tm_g_bivariate()
+
+  module_ns <- app$active_module_ns()
+  row_facet_selector <- sprintf("#%s-row_facet-dataset_CO2_singleextract-select", module_ns)
+  col_facet_selector <- sprintf("#%s-col_facet-dataset_CO2_singleextract-select", module_ns)
+  x_scale_selector <- sprintf("#%s-free_x_scales", module_ns)
+  y_scale_selector <- sprintf("#%s-free_y_scales", module_ns)
```

Comment:
@vedhav I think we have `app$active_module_element()` for that

```r
> app$active_module_element
function(element) {
      checkmate::assert_string(element)
      sprintf("#%s-%s", self$active_module_ns(), element)
    }
<environment: 0x000001fca26a01e0>
```


So instead of 

```r
module_ns <- app$active_module_ns()
  row_facet_selector <- sprintf("#%s-row_facet-dataset_CO2_singleextract-select", module_ns)
```

you can do 

```r
row_facet_selector <- app$active_module_element("row_facet-dataset_CO2_singleextract-select")
```

Reply:
thanks

Reply:
did that here 8a5b7cbf7b18ec4e67c578ca2885329ce4c4bc94